.github/workflows/tests.yml

```diff
@@ -99,6 +99,9 @@ jobs:
           PR_NUMBER: ${{ github.event.pull_request.number }}
         run: echo "TRAVIS_PULL_REQUEST=${PR_NUMBER:-false}" >> $GITHUB_ENV
 
+      - name: Install Clippy
```

Comment:
Right now this is under the `compilation` key, which runs it on all steps in the matrix. I don't think that's necessary. 

Probably the best solution is adding it to its own build job, kind of like the benchmarking on nightly. We might in the future decide that we want to run it on its own matrix, so we can get warnings when new lints are introduced on the beta channel before they hit stable.

Reply:
OK. I just went ahead and added the matrix. stable and beta.

Reply:
... which makes me realise and I never actually removed this Install clippy item. and the fact that Clippy still succeeded despite that means that it's being installed by default anyway.